apax/md/simulate.py

```diff
@@ -208,31 +226,47 @@ def run_sim(
     pbar_update_freq = int(np.ceil(500 / n_inner))
     pbar_increment = n_inner * pbar_update_freq
 
+    sampling_rate = traj_handler.sampling_rate
+    on_eval, no_eval = create_evaluation_functions(
+        sim_fns.auxiliary_fn,
+        state.position,
+        system.atomic_numbers,
+        neighbor,
+        system.box,
+    )
+
     @jax.jit
-    def sim(state, neighbor):  # TODO make more modular
+    def sim(state, outer_step, neighbor):  # TODO make more modular
         def body_fn(i, state):
-            state, neighbor = state
+            state, outer_step, neighbor = state  # neighbors
             if isinstance(state, simulate.NPTNoseHooverState):
                 box = state.box
                 apply_fn_kwargs = {}
             else:
                 box = system.box
-                apply_fn_kwargs = {"box": box}
+                apply_fn_kwargs = {"box": box}  # this might cause a bug
```

Comment:
should we discuss this bevor merging?

Reply:
The comment was just a reminder for myself to look into this. it's all working as intended. I'll remove the comment